blosc2/c2array.py

```diff
@@ -9,14 +9,84 @@
 from __future__ import annotations
 
 from collections.abc import Sequence
+from contextlib import contextmanager
+import os
 
 import httpx
 import numpy as np
 
 import blosc2
 
 
+C2SUB_DEFBASE_ENVVAR = 'CATERVA2_SUBSCRIBER_URL'
```

Comment:
Renamed the code variable in commit 2a71810. IMO the environment variable on the other hand should be readable outside of Python code (that is, in the environment) and more descriptive. Also, Blosc2 users may know nothing about the `cat2sub` or `cat2pub` programs run in servers. (Tangentially, Caterva2 has other similarly named variables as `CATERVA2_AUTH_SECRET` which do not match its internal code either.)

Reply:
It may make sense though to prefix it with `BLOSC2_` since it's for the use of the Blosc2 library. Maybe `BLOSC2_CATERVA2_SUBSCRIBER`? (Maybe `BLOSC2_CAT2SUB_URL`, though it looks cryptic in the environment.)

Reply:
I am not a big fan of having long envvars. For me, `CATERVA2_URLBASE` would be a good compromise (I like `URLBASE` suffix because it is already used internally in doc strings, so the meaning is pretty clear).

Reply:
Umm, what about then having it describe its purpose instead? Something like `BLOSC2_C2ARRAY_URLBASE`.

Reply:
Ok. Deal.

Reply:
Will use `BLOSC_C2URLBASE` for consistency and compactness.